github/ci/services/common/k8s/pkg/client/client.go

```diff
@@ -0,0 +1,39 @@
+package client
+
+import (
+	"os/user"
+	"path"
+
+	certmanagerclientset "github.com/jetstack/cert-manager/pkg/client/clientset/versioned"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func NewClientset() (*kubernetes.Clientset, error) {
+	config, err := getConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	return kubernetes.NewForConfig(config)
+}
+
+func NewCertManagerClientset() (*certmanagerclientset.Clientset, error) {
+	config, err := getConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	return certmanagerclientset.NewForConfig(config)
+}
+
+func getConfig() (*restclient.Config, error) {
```

Comment:
Is this in sync with what the gitops rules to to get the kubeconfig?

Reply:
yes, by default gitops rules works with `~/.kube/config` and it can be configured also with env vars according to https://github.com/adobe/rules_gitops/blob/d5eb2b90c7b107eb6c91134fa4f21ea97ac15981/skylib/k8s.bzl#L285-L381 we can always modify our client code to get the credentials from other locations via flags or from env vars if needed.